File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    const minPressMap = {};

    for (const keys of keymap) {
      for (let i = 0; i < keys.length; i++) {
        const char = keys[i];
        const pressCount = i + 1;

        if (!minPressMap[char] || pressCount < minPressMap[char]) {
          minPressMap[char] = pressCount;
        }
      }
    }
    
    return targets.map(word => {
        return word.split("").reduce((acc, char) => {
            if (acc === -1 || !minPressMap[char]) return -1;
            return acc + minPressMap[char];
        }, 0);
    });
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    min_press = {}
    
    for kmap in keymap:
        for idx, char in enumerate(kmap):
            press_count = idx + 1

            if char in min_press:
                min_press[char] = min(min_press[char], press_count)
            else:
                min_press[char] = press_count
    
    for target in targets:
        total_press = 0
        possible = True
        
        for char in target:
            if char in min_press:
                total_press += min_press[char]
            else:
                possible = False
                break
        
        if possible:
            answer.append(total_press)
        else:
            answer.append(-1)
    
    return answer